apps/src/lib/client/rpc.rs

```diff
@@ -40,8 +41,66 @@ use crate::facade::tendermint_rpc::query::Query;
 use crate::facade::tendermint_rpc::{
     Client, HttpClient, Order, SubscriptionClient, WebSocketClient,
 };
+use crate::node::ledger::events::Event;
 use crate::node::ledger::rpc::Path;
 
+/// Query the status of a given transaction.
+///
+/// If a response is not delivered until `deadline`, we exit the cli with an
+/// error.
+pub async fn query_tx_status(
+    status: TxEventQuery<'_>,
+    args: args::Query,
+    deadline: Instant,
+) -> Event {
+    tokio::time::timeout_at(deadline, async move {
+        let client = HttpClient::new(args.ledger_address).unwrap();
+
+        const ONE_SECOND: Duration = Duration::from_secs(1);
+        let mut backoff = ONE_SECOND;
+
+        loop {
+            let data = vec![];
+            let response = client
+                .abci_query(Some(status.into()), data, None, false)
+                .await
+                .unwrap();
+            let mut events = match response.code {
+                Code::Ok => {
+                    match Vec::<Event>::try_from_slice(&response.value[..]) {
+                        Ok(events) => events,
+                        Err(err) => {
+                            eprintln!("Error decoding the event value: {err}");
+                            break Err(());
+                        }
+                    }
+                }
+                Code::Err(err) => {
+                    eprintln!(
+                        "Error in the query {} (error code {})",
+                        response.info, err
+                    );
+                    break Err(());
+                }
+            };
+            if let Some(e) = events.pop() {
+                // we should only have one event matching the query
+                break Ok(e);
+            }
+            // simple linear backoff - if an event is not available,
+            // increase the backoff duration by one second
```

Comment:
Could indicate we are doing a the retry to the user somewhere in this loop e.g. something like

```rust
println!("No events received, may retry in $backoff seconds")
```

Reply:
I think we normally take longer than a second to get a response. So we might want to change the backoff or else we will always print messages that make people worry :sweat_smile: